controllers/flinkcluster_reconciler.go

```diff
@@ -403,20 +403,34 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 
 	// Delete
 	if desiredJob == nil && observedJob != nil {
-		if len(jobID) > 0 {
+		// Cancel Flink job if it is live
+		// 1) In the case of which savepoint was triggered, after it is completed, proceed to delete step.
+		// 2) When savepoint was skipped, continue to next step immediately.
+		//
+		// If savepoint or cancellation was failed, the control state will fall to the failed in the updater.
+		if len(jobID) > 0 && len(observed.flinkRunningJobIDs) == 1 {
 			log.Info("Cancelling job", "jobID", jobID)
-			var err = reconciler.cancelFlinkJob(jobID, true /* takeSavepoint */)
+			if isSavepointTriggered(reconciler.observed.cluster.Status.Savepoint) {
```

Comment:
What if the savepoint has completed (instead of triggered an in-progress) at this point? Seems it will go ahead and call `cancelFlinkJobAsync` again?

Reply:
Yes, it proceeds to call cancelFlinkJobAsync.